src/goto-symex/slice.cpp

```diff
@@ -112,7 +112,7 @@ void symex_slicet::slice(symex_target_equationt::SSA_stept &SSA_step)
 void symex_slicet::slice_assignment(
   symex_target_equationt::SSA_stept &SSA_step)
 {
-  assert(SSA_step.ssa_lhs.id()==ID_symbol);
+  PRECONDITION(SSA_step.ssa_lhs.id() == ID_symbol);
```

Comment:
I believe this should be true by design.

Reply:
Still, isn't it a good idea to leave it there, even if only for documentation purposes?

Reply:
Not a big deal, but I think if we were to go down that route we'd have, e.g., any use of an `index_exprt` prefixed by `expr.id() == ID_index`.